File: university/src/main/java/com/gutorov/university/entity/BaseEntity.java
package com.gutorov.university.entity;

import java.util.UUID;

public abstract class BaseEntity {
    protected UUID id;

    protected BaseEntity() {
    }

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }
}

File: src/main/java/com/gutorov/university/entity/BaseEntity.kt
package com.gutorov.university.entity

import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id
import jakarta.persistence.MappedSuperclass
import java.util.*

@MappedSuperclass
abstract class BaseEntity {
    @Id
    @GeneratedValue(generator = "uuid2")
    open var id: UUID = UUID.randomUUID()
        protected set

    protected constructor()
}